core/src/main/java/com/github/swissquote/carnotzet/core/Carnotzet.java

```diff
@@ -116,9 +116,11 @@ public Carnotzet(CarnotzetConfig config) {
 	public List<CarnotzetModule> getModules() {
 		if (modules == null) {
 			modules = resolver.resolve(config.getTopLevelModuleId(), failOnDependencyCycle);
-			if (SystemUtils.IS_OS_LINUX || !getResourcesFolder().resolve("expanded-jars").toFile().exists()) {
+			if (!SystemUtils.IS_OS_WINDOWS || !getResourcesFolder().resolve("expanded-jars").toFile().exists()) {
 				resourceManager.extractResources(modules);
-				modules = computeServiceIds(modules);
+			}
+			modules = computeServiceIds(modules);
+			if (!SystemUtils.IS_OS_WINDOWS || !getResourcesFolder().resolve("expanded-jars").toFile().exists()) {
```

Comment:
Doing the same check twice seems inefficient, why not store it in a variable instead ? would also clarify intent of the check

Reply:
Which check?